src/tagstore.rs

```diff
@@ -0,0 +1,111 @@
+use serde::de::{Deserializer, Error};
+use serde::ser::{Serialize, Serializer};
+
+/// signals that a newtype is from a CBOR tag
+pub const CBOR_NEWTYPE_NAME: &str = "__cbor_tag";
```

Comment:
This should include some special char to make sure this is *never* the actual name of the struct. For example include `\0` or `\x1b` at the start.